src/Cms/Page.php

```diff
@@ -1401,7 +1401,9 @@ public function slug(string $languageCode = null): string
                 $languageCode = $this->kirby()->languageCode();
             }
 
-            if ($translation = $this->translations()->find($languageCode)) {
+            $defaultLanguageCode = $this->kirby()->defaultLanguage()->code();
+
+            if ($languageCode != $defaultLanguageCode && $translation = $this->translations()->find($languageCode)) {
```

Comment:
```suggestion
            if ($languageCode !== $defaultLanguageCode && $translation = $this->translations()->find($languageCode)) {
```

One more unnecessary query, great! Using identical operator will be more stable 👍